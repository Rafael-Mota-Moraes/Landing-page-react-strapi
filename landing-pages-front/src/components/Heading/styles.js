import styled, { css } from "styled-components";

const titleSize = (size, theme) => {
  if (size === "small") {
    return css`
      font-size: ${theme.font.sizes.medium};
    `;
  }

  if (size === "medium") {
    return css`
      font-size: ${theme.font.sizes.large};
    `;
  }

  if (size === "big") {
    return css`
      font-size: ${theme.font.sizes.xlarge};
    `;
  }

  if (size === "huge") {
    return css`
      font-size: ${theme.font.sizes.xhuge};
      ${mediaFont(theme)}
    `;
  }
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.xlarge};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? "uppercase" : "none"};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize(size, theme)};
    ${titleCase(uppercase)};
  `}
`;
