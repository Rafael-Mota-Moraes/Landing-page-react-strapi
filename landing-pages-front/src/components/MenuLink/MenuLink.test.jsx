import { screen } from "@storybook/testing-library";
import { MenuLink } from ".";
import { renderTheme } from "../../styles/renderTheme";

describe("<MenuLink />", () => {
  it("should render a link", () => {
    renderTheme(<MenuLink link="http://localhost:3000">Children</MenuLink>);

    expect(screen.getByRole("link", { name: "Children" })).toHaveAttribute(
      "target",
      "_self"
    );
  });

  it("should open in a new tab", () => {
    renderTheme(
      <MenuLink link="http://localhost:3000" newTab={true}>
        Children
      </MenuLink>
    );

    expect(screen.getByRole("link", { name: "Children" })).toHaveAttribute(
      "target",
      "_blank"
    );
  });
  it("should match a snapshot", () => {
    renderTheme(
      <MenuLink link="http://localhost:3000" newTab={false}>
        Children
      </MenuLink>
    );

    expect(screen.getByRole("link", { name: "Children" })).toMatchSnapshot();
  });
});
