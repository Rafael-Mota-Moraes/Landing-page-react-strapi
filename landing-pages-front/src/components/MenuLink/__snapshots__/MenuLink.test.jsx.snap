// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should match a snapshot 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 2.4rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="http://localhost:3000"
  target="_self"
>
  Children
</a>
`;

exports[`<MenuLink /> should open in a new tab 1`] = `
.c0 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 2.4rem;
  color: #0a1128;
  position: relative;
}

.c0::after {
  content: "";
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c0:hover::after {
  left: 25%;
  width: 50%;
}

<a
  class="c0"
  href="http://localhost:3000"
  target="_self"
>
  Children
</a>
`;
